Add note about '/liveness' endpoint for standalone agent (#1090)
elastic/ingest-docs: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-monitoring.asciidoc

collected. If neither setting is specified, monitoring is turned off. Set
`use_output` to specify the output to which monitoring events are sent.

> Setting `agent.monitoring.enabled` to `true` also exposes a `/liveness` endpoint. By default, the endpoint 

Reviewer: It's actually `agent.monitoring.enabled.http` that needs to be enabled, `agent.monitoring.enabled` controls whether we collect monitoring data from agent to be shipped to Fleet.

The liveness probe is an HTTP feature. Sorry for the late review.

Author: Thanks for spotting this @cmacknz!

A couple of questions:
 - Should it be `agent.monitoring.enabled.http` or `agent.monitoring.http.enabled`? In your [comment](https://github.com/elastic/elastic-agent/issues/107#issuecomment-2133910470) you mentioned the latter.
 - I don't think the setting is available to in Fleet, so I think I should mention the setting only in the standalone agent docs ([here](https://www.elastic.co/guide/en/fleet/8.13/elastic-agent-monitoring-configuration.html)). Does that seem right to you?

collected. If neither setting is specified, monitoring is turned off. Set
`use_output` to specify the output to which monitoring events are sent.

> Setting `agent.monitoring.enabled` to `true` also enables a `/liveness` endpoint to be 
> available where that server is configured to be exposed. By default, the endpoint 
> returns a `200` OK status as long as {agent}'s internal main loop is responsive and can 
> process configuration changes. It can be configured to also monitor the component states 
> to return an error if anything is degraded or failed.

Reviewer 2: "where that server is configured to be exposed" is a little confusing to me - does that just mean at the URL for the server? Would that be different for `/debug/pprof/` and `/buffer`? if not, I'd probably skip the "where configured" bit

```suggestion
Setting `agent.monitoring.enabled` to `true` also exposes a `/liveness` endpoint. By default, the endpoint 
returns a `200` OK status as long as {agent}'s internal main loop is responsive and can 
process configuration changes. It can be configured to also monitor the component states 
and return an error if anything is degraded or has failed.
```
